core/src/main/java/org/kiwix/kiwixmobile/core/search/viewmodel/effects/ShowToast.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2020 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.core.search.viewmodel.effects
+
+import android.widget.Toast
+import androidx.annotation.StringRes
+import androidx.appcompat.app.AppCompatActivity
+import org.kiwix.kiwixmobile.core.base.SideEffect
+import org.kiwix.kiwixmobile.core.extensions.toast
+
+data class ShowToast(@StringRes val stringId: Int) : SideEffect<Unit> {
```

Comment:
I'm not able to understand why do we need a data class just to show a toast. :(
Can't we use extension functions?

Reply:
Because it is a side effect, the viewmodel does not have an activity because a viewmodel survives configurations and therefore multiple activities, we use the extension function in this class. The benefits of Unidirectional Data flow is an incredibly dumb view that simply renders states and forwards events, this viewmodel handles these events then either emits a new state or instructs a side effect to happen